M2/Macaulay2/m2/installPackage.m2

```diff
@@ -634,8 +666,10 @@ installPackage Package := opts -> pkg -> (
     verboseLog := if opts.Verbose or debugLevel > 0 then printerr else identity;
 
     use pkg;
-    -- TODO: make this more functional
-    chkdoc = opts.CheckDocumentation;			    -- oops, this will have a lingering effect...
+    if chkdoc then checkForErrors pkg;
```

Comment:
Why was this call to `checkForErrors` added at the beginning of `installPackage`?  We haven't reset the counters yet, so it might complain about an old, out-of-date error.

Quick example:  Suppose we have the following silly package:

```m2
newPackage("Foo", Headline => "foo",)
beginDocumentation()

doc ///
  Key
    Foo
  Description
    Example
      1/0
///
```
Of course, the first call to `installPackage` fails.  So we go back and change that `1/0` to, say, `1/2`, save the file, and try again.  But it still thinks there's an error:

```m2
i2 : installPackage("Foo", FileName => "Foo.m2")
stdio:2:14:(3): error: installPackage: 1 error(s) occurred in running examples for package Foo
```
If we comment out the call to `checkForErrors` here, then the second call to `installPackage` works as expected -- it tries running the example again, and this time it's successful.

Reply:
I think the idea was to signal silent documentation errors from `loadPackage`.

> So we go back and change that 1/0 to, say, 1/2, save the file, and try again.

I don't think there's any guarantee that this would work. e.g. if the package called `needs` inside, the second time they might not be loaded at all. It would be good to get it to work, but there are numerous points of failure.